tailcfg/tailcfg.go

```diff
@@ -975,3 +974,28 @@ type WhoIsResponse struct {
 	Node        *Node
 	UserProfile *UserProfile
 }
+
+// Oauth2Token is a copy of golang.org/x/oauth2.Token, to avoid the
+// go.mod dependency on App Engine and grpc, which was causing problems.
+// All we actually needed was this struct on the client side.
+type Oauth2Token struct {
```

Comment:
Do any of the rendering methods off of the oauth2.Token type matter? E.g. String?

Reply:
Empirically no? I was able to "tailscale up" and log in with this.

Reply:
Traced all the code, it ultimately just ends up here:

```go
        request := tailcfg.RegisterRequest{
                Version:    1,
                OldNodeKey: tailcfg.NodeKey(oldNodeKey),
                NodeKey:    tailcfg.NodeKey(tryingNewKey.Public()),
                Hostinfo:   hostinfo,
                Followup:   url,
        }
        c.logf("RegisterReq: onode=%v node=%v fup=%v",
                request.OldNodeKey.ShortString(),
                request.NodeKey.ShortString(), url != "")
        request.Auth.Oauth2Token = t
        request.Auth.Provider = persist.Provider
        request.Auth.LoginName = persist.LoginName
        request.Auth.AuthKey = authKey
        bodyData, err := encode(request, &serverKey, &c.machinePrivKey)
```

... using normal reflect-based JSON marshalling with the struct tags.